sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -2127,4 +2127,243 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
       assert(df.schema === expectedSchema)
     }
   }
+
+  def testFile(fileName: String): String = {
+    Thread.currentThread().getContextClassLoader.getResource(fileName).toString
```

Comment:
Just a question, is it hard to do this with `Files.write(path.toPath, data.getBytes(StandardCharsets.UTF_8))`? I thought it makes other tests easily added in the future. If it needs many changes, I am fine as is too.

Reply:
Please, explain what do you mean. The method returns full path to a test, you code writes something in UTF-8. Or do you mean replacing test files by forming them in code?